oauth-1.0a.js

```diff
@@ -95,6 +99,16 @@ OAuth.prototype.getSignature = function(request, token_secret, oauth_data) {
 };
 
 /**
+ * Create a OAuth Body Hash
+ * @param {Object} request data
+ */
+OAuth.prototype.getBodyHash = function(request, token_secret) {
+  var body = typeof request.data === 'string' ? request.data : JSON.stringify(request.data)
+
+  return this.hash_function(body, this.getSigningKey(token_secret))
```

Comment:
Assuming hmac-sha1 signature, the body hash should sha1.

E.g. https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#anchor15